src/Cms/Find.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Kirby\Cms;
+
+use Kirby\Exception\InvalidArgumentException;
+use Kirby\Exception\NotFoundException;
+use Kirby\Toolkit\Str;
+
+/**
+ * The Find class is used in the API and
+ * the Panel to find models and parents
+ * based on request paths
+ *
+ * @package   Kirby Cms
+ * @author    Bastian Allgeier <bastian@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier GmbH
+ * @license   https://getkirby.com/license
+ */
+class Find
```

Comment:
Maybe the name should be less generic and more tied to the objects it treats, e.g. `ModelFind` or so

Reply:
Like `Finder`?

Reply:
No, `Finder` is just as generic as `Find` - I think it would be good to create a connection to models with the name. Having `Kirby\Cms\Find` next to `Kirby\Cms\Search` is just so similar.

Reply:
I wanted a name that is "fun" to use in Area and API routes. I'm not 100% sure about Find but at least it really looks nice and readable in routes. We will have those methods all over the place. They will be in the dialogs as well.

Reply:
What about `Access` to convey the concept of trying to access those models in routes?

Reply:
`Access` is worse cause it's all about getting the model not what access we have to them. `find` is good as the concept what it is achieving. And I get the part of super short and fun names to use, but with so many classes we have I'm also concerned about just looking at a list of classes and knowing what each one does.

Reply:
I guess it'd be easier if we had a `Kirby\Model` namespace :D

Reply:
@distantnative I agree, but we open yet another one of Pandora's boxes with that. I have no better idea to be honest. `ModelFinder::page()` is really something I don't want to use in the routes. In the API we already solve this by binding the API instance to routes and then routes can access models with $this->page(), $this->file() etc. But I have to say that this obscure binding of instances to $this doesn't feel very good either. I don't want to follow the same path for Fiber routes and also would love to move away from that in Api routes some day.

Reply:
Ok, I won't block this over the name 😅😉

Reply:
Thanks :) Maybe our Greenland shark @lukasbestle should still have a look at it and give his opinion. The biggest problem is still that we are stuck with the name once we introduce the class.